python/pyiceberg/catalog/__init__.py

```diff
@@ -323,6 +323,17 @@ def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: U
             NoSuchTableError: If a table with the name does not exist
         """
 
+    @abstractmethod
+    def update_table(self, identifier: Union[str, Identifier], updates: Tuple[TableUpdate, ...]) -> Table:
```

Comment:
FYI, we're adding a similar method in @nastra's transaction work: https://github.com/apache/iceberg/pull/7569

We probably want to align the naming so maybe this should be `commit_table` to align with the `commit_transaction`.

That work is also updating the request to have an optional identifier, so you could just make the identifier part of the `TableUpdate` in Python.

Reply:
Nice one, thanks!